py-development.Rmd

```diff
@@ -0,0 +1,540 @@
+step-by-step instructions to get up and running with `conda` follow.
+
+1. To install a new Python package from the Anaconda repositories, simply run
+   `conda install <package name>` in a terminal. You can also use the `pip`
+   package manager, but it will be easier to keep track of packages by sticking
+   to one installation method.
+2. Some packages are not available in the default Anaconda repositories. User
+   contributed packaged are available in Anaconda "channels", use `anaconda
+   search -t conda <package name>`, to find a channel with the desired package.
+   To install this package, use `conda install -c <channel name> <package
+   name>`. The [conda forge channel](https://conda-forge.github.io/) channel
+   has many of the packages not in the default repositories.
+
+## How to get help online {#py-dev-how-to-get-help-online}
+
+When reading the built-in Python help is not sufficient, there are several 
+online resources that can helpful. One of the most commonly used resources for
+data science is the Stack Exchange network wich offer Q&A sites both for
+programming related topics via [Stack Overflow][so], as well as statistics and
+machine learning via [Cross Validated][cv].
+
+If nothing relevant can be found after searching the many existing question and
+answers on these sites, it is appropriate to ask a new question! Stack Overflow
+has detailed instruction on how to create a [minimal reproducible
+example][so-mre] when asking a question to increase the chances that the
+question receives a specific and helpful answer. The key principles listed on
+the website recommends that and answer follows these guidelines: 
+
+> - Minimal – Use as little code as possible that still produces the same
+> problem
+> - Complete – Provide all parts someone else needs to reproduce your problem
+> in the question itself
+> - Reproducible – Test the code you're about to provide to make sure it
+> reproduces the problem
+
+An additional benefit of reducing the problem into this format is that you
+might discover the error in the process! This process helps narrowing down
+exactly which region of the code is failing and the act of explaining the
+problem often reveals a solution before any replies have come in (commonly
+referred to as [rubber duck debugging][wiki-rubber-duck-debugging].
+
+## Key Points {#py-dev-keypoints}
+
+```{r child="keypoints/py-development.md"}
+```
+
+## Exercises {#py-dev-exercises}
+
+1. Write a function that returns the mean of exactly two numbers.
+2. Write a function that returns the mean of any amount of numbers.
+3. Write function that takes a string as its input and returns the
+   first and last character of the string.
+4. Write a function that returns any character from the string. Which character
+   is returned should be determined by an integer specified via a named
+   parameter that has the default value of `0`.
+5. Expand on the previous function by include a assert statement that checks if
```

Comment:
If this is a multi-part exercise, make it one exercise with two parts, not two exercises that are linked.